src/amo/components/AddAddonToCollection/index.js

```diff
@@ -156,6 +156,10 @@ export class AddAddonToCollectionBase extends React.Component<InternalProps> {
     );
   }
 
+  getBlankName(): string {
+    return this.props.i18n.gettext('(no name)');
```

Comment:
Given that it could happen everywhere we isplay a collection and that it's a placeholder to avoid more problems, I am wondering if we shouldn't just use a default value in the reducer. We wouldn't have i18n probably but I don't think it'd be a problem and that would aboid repeating this fix everywhere we need the collection's name.

Reply:
That was my first approach, but, as you said, it means we lose i18n. I didn't think that was something we would want to lose, as we're pretty careful about that pretty much everywhere else in the site.

@jvillalobos @diox What do you think? Would be be acceptable to output "(no name)", in English, every time we need to display the name of a collection that has a missing name, regardless of which language the rest of the site is in?

Reply:
I would prefer that it's localized, but it should also be a rare issue. If really necessary, I'm okay with hardcoding it in English.

Reply:
I don't think it's a big issue. It just means we duplicate the code for `i18n.gettext('(no name)')` in a few places, but I don't see that as such a big problem that we should sacrifice `i18n` for it.

Reply:
> I don't think it's a big issue. It just means we duplicate the code for `i18n.gettext('(no name)')` in a few places, but I don't see that as such a big problem that we should sacrifice `i18n` for it.

I disagree. It means we have to fix this problem every time we want to display a collection name. The fact that it is the second patch for the same thing speaks for itself.

If we want to keep i18n, fine, but we should have a helper function similar to what we've done here: https://github.com/mozilla/addons-frontend/blob/7c535911a9d3a650cf829166aa098603dc6edc3c/src/amo/reducers/versions.js#L336

Ideally we'd have a way to ensure that the getter will be called instead of the property 🤷

Reply:
You're not wrong, about the helper function, but that adds even more overhead, both in terms of creating the function, and having to remember to call it each time. You also have a point about this being the second patch, but that was just me being lazy and not making sure all cases were covered with the first patch. 

I'm not really sure it's worth doing the extra work for the helper, but it is closer to what I would have liked (which was to have this in the reducer), so I'll make that change and update the patch accordingly.

Reply:
Ok, I have landed the changes in a new commit @willdurand.